tests/integration_tests/style/test_json.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests for JSON linter checks."""
+
+import framework.utils as utils
+import json
+
+
+def test_json_style():
+    """Fail if the repository contains malformed JSON in .json files"""
+
+    # Get all *.json files from the project
+    json_files = utils.get_files_from(
+        find_path="..",
+        pattern="*.json",
+        exclude_names=["build"])
+
+    assert len(json_files) != 0
+
+    # for each .json file we find, check that
+    # it can be parsed as JSON
+    for file_name in json_files:
```

Comment:
I see a little problem here. If multiple files are invalid JSONs the test will fail upon reaching the first invalid JSON. This can be a burden for the contributors since they will not see all the failing JSONs from a one shot run of the test. I suggest to modify the `is_json_file_valid ` function into a function that goes through all the files and appends to a string all the paths of the invalid JSONs. Something like this:
```
def _validate_files(json_files):
    error_msgs = ""
    # for each .json file we find, check that
    # it can be parsed as JSON
    for file_name in json_files:
        with open(file_name, 'r') as file_stream:
            try:
                json.load(file_stream)
            except json.JSONDecodeError:
                if error_msgs:
                    error_msgs += ","
                # json failed to decode
                error_msgs += "{} ".format(file_name)
    return error_msgs
```
Then in the `test_json_style` function you could have:
```
    error_msgs = _validate_files(json_files)
    if error_msgs:
        assert False, "Files {}contain invalid JSON".format(str(error_msgs))
```
WDYT?

Reply:
@dianpopa Oh yeah that's an excellent point, I'll go ahead and implement that.